### PR TITLE
Fix bundle merge reveal check

### PR DESCRIPTION
**Pull request by crisdut**
**Description**

This PR intent fix the merge reveal bundle check. Also, adjusts the error message for correct naming.

### CODE REVIEW

File: src/accessors/merge_reveal.rs [hunk 1]
```diff
@@ -187,12 +187,12 @@ impl<Seal: ExposedSeal> MergeReveal for Assignments<Seal> {
 impl MergeReveal for TransitionBundle {
     fn merge_reveal(mut self, other: Self) -> Result<Self, MergeRevealError> {
         debug_assert_eq!(self.commitment_id(), other.commitment_id());
-        if self.known_transitions.len() + other.known_transitions.len() > self.input_map.len() ||
+        if self.input_map.len() < other.known_transitions.len() ||
```
Comment (dr-orlovsky) [inline src/accessors/merge_reveal.rs:190]: I think we need to take into account both `self` and `other` when do comparison. Why do you think the original version was buggy?

Comment (dr-orlovsky) [reply src/accessors/merge_reveal.rs:190]: All right, I see. I think it should be this way: first we do `extend`, than we do check

Comment (crisdut) [reply src/accessors/merge_reveal.rs]: I imagined that the original version made a purposeful comparison between `self` and `other`, so I adjusted it to a verification that made more sense.

So, should we first do the `extend` and then check if the amount of `input_map` is less than the `known_transitions` ?

Comment (dr-orlovsky) [reply src/accessors/merge_reveal.rs]: Yes

Comment (crisdut) [reply src/accessors/merge_reveal.rs]: Done

**Review by dr-orlovsky — APPROVED** [review]
ACK https://github.com/RGB-WG/rgb-std/commit/d53454a5c6c440b163b7eb399aa276049dc6a08a

**Pull request merged**